idb/postgres/postgres.go

```diff
@@ -2440,16 +2444,29 @@ func (db *IndexerDb) buildAccountQuery(opts idb.AccountQueryOptions) (query stri
 	// TODO: asset holdings and asset params are optional, but practically always used. Either make them actually always on, or make app-global and app-local clauses also optional (they are currently always on).
 	withClauses = append(withClauses, "qaccounts AS ("+query+")")
 	query = "WITH " + strings.Join(withClauses, ", ")
-	if opts.IncludeAssetHoldings {
-		query += `, qaa AS (SELECT xa.addr, json_agg(aa.assetid) as haid, json_agg(aa.amount) as hamt, json_agg(aa.frozen) as hf, json_agg(aa.created_at) as holding_created_at, json_agg(aa.closed_at) as holding_closed_at, json_agg(aa.deleted) as holding_deleted FROM account_asset aa JOIN qaccounts xa ON aa.addr = xa.addr GROUP BY 1)`
-	}
-	if opts.IncludeAssetParams {
-		query += `, qap AS (SELECT ya.addr, json_agg(ap.index) as paid, json_agg(ap.params) as pp, json_agg(ap.created_at) as asset_created_at, json_agg(ap.closed_at) as asset_closed_at, json_agg(ap.deleted) as asset_deleted FROM asset ap JOIN qaccounts ya ON ap.creator_addr = ya.addr GROUP BY 1)`
+	if opts.IncludeDeleted {
```

Comment:
Toggles whether `WHERE coalesce(___.deleted, false) = false` is in the query or not. This is already a little out of hand, we need to be careful about what we add here.

Should default the account endpoint to excluding deleted things with no option to include them? The creatable-specific endpoints could have the option and the implementation would be cleaner.

Reply:
This is getting a bit nuts. Would it be easier to client-side-filter some of these things for deletion? Is it weird that IncludeDeleted becomes recursive on an account query and is applied to whether sub-fields of an account are deleted or not?